ogr/ogrsf_frmts/miramon/mm_gdal_functions.c

```diff
@@ -0,0 +1,2801 @@
+            }
+            if (*ptr1 == cometa)
+                *(ptr1 - 1) = 0;
+            else
+                *ptr1 = 0;
+        }
+    }
+    return cadena;
+} /* End of MM_RemoveInitial_and_FinalQuotationMarks() */
+
+char *MM_RemoveLeadingWhitespaceOfString(char *cadena)
+{
+    char *ptr;
+    char *ptr2;
+
+    if (cadena == nullptr)
+        return cadena;
+
+    for (ptr = cadena; *ptr && (*ptr == ' ' || *ptr == '\t'); ptr++)
+        continue;
+
+    if (ptr != cadena)
+    {
+        ptr2 = cadena;
+        while (*ptr)
+        {
+            *ptr2 = *ptr;
+            ptr2++;
+            ptr++;
+        }
+        *ptr2 = 0;
+    }
+    return cadena;
+}
+
+char *MM_RemoveWhitespacesFromEndOfString(char *str)
+{
+    const char *s;
+
+    if (str == nullptr)
+        return str;
+
+    for (s = str; *s; ++s)
+        continue;
+    return MM_l_RemoveWhitespacesFromEndOfString(str, (s - str));
+}
+
+struct MM_ID_GRAFIC_MULTIPLE_RECORD *
+MMCreateExtendedDBFIndex(FILE_TYPE *f, MM_EXT_DBF_N_RECORDS nNumberOfRecords,
+                         MM_FIRST_RECORD_OFFSET_TYPE offset_1era,
+                         MM_ACUMULATED_BYTES_TYPE_DBF bytes_per_fitxa,
+                         MM_ACUMULATED_BYTES_TYPE_DBF bytes_acumulats_id_grafic,
+                         MM_BYTES_PER_FIELD_TYPE_DBF bytes_id_grafic,
+                         MM_BOOLEAN *isListField, MM_EXT_DBF_N_RECORDS *nMaxN)
+{
+    struct MM_ID_GRAFIC_MULTIPLE_RECORD *id;
+    MM_EXT_DBF_N_RECORDS i_dbf;
+    MM_EXT_DBF_SIGNED_N_RECORDS i, id_grafic;
+    char *fitxa;
+    MM_BYTES_PER_FIELD_TYPE_DBF bytes_final_id_principi_id1 =
+        bytes_per_fitxa - bytes_id_grafic;
+
+    *isListField = FALSE;
+    *nMaxN = 0;
+    if (!nNumberOfRecords)
+        return nullptr;  // No elements to read
+
+    if (MMCheckSize_t(nNumberOfRecords * sizeof(*id), 1))
+        return nullptr;
+    if (nullptr == (id = (struct MM_ID_GRAFIC_MULTIPLE_RECORD *)calloc_function(
+                        (size_t)nNumberOfRecords * sizeof(*id))))
+        return nullptr;
+
+    if (MMCheckSize_t(bytes_id_grafic + 1, 1))
+        return nullptr;
+    if (nullptr ==
+        (fitxa = (char *)calloc_function((size_t)bytes_id_grafic + 1)))
+    {
+        free_function(id);
+        return nullptr;
+    }
+    fitxa[bytes_id_grafic] = '\0';
+
+    fseek_function(f,
+                   (MM_FILE_OFFSET)offset_1era +
+                       (MM_FILE_OFFSET)bytes_acumulats_id_grafic,
+                   SEEK_SET);
+
+    i_dbf = 0;
+    do
+    {
+        if (i_dbf == nNumberOfRecords ||
+            fread_function(fitxa, 1, bytes_id_grafic, f) !=
+                (size_t)bytes_id_grafic)
+        {
+            free_function(id);
+            free_function(fitxa);
+            return nullptr;
+        }
+        i_dbf++;
+    } while (1 !=
+                 sscanf(fitxa, scanf_MM_EXT_DBF_SIGNED_N_RECORDS, &id_grafic) ||
+             id_grafic < 0);
+    i = 0;
+
+#ifdef _MSC_VER
```

Comment:
ah, so MSVC doesn't like while (TRUE) at all in full warning mode. Ok, so forget my previous comments about that